### PR TITLE
Add a common cloudprovider observer

### PR DESCRIPTION
**Pull request by gnufied**
cc @deads2k

### CODE REVIEW

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+type cloudProviderObserver struct {
+	targetNamespaceName string
+}
+
+// ObserveCloudProviderNames observes the cloud provider from the global cluster infrastructure resource.
+func (cloud *cloudProviderObserver) ObserveCloudProviderNames(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(InfrastructureLister)
+	var errs []error
+	cloudProvidersPath := []string{"extendedArguments", "cloud-provider"}
```
Comment (deads2k) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: this path needs to be a parameter

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	cloudProviderConfPath := []string{"extendedArguments", "cloud-config"}
```
Comment (deads2k) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: this path needs to be a parameter

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+// NewCloudProviderObserver returns a new cloudprovider observer for syncing cloud provider specific
+// information to controller-manager and api-server.
+func NewCloudProviderObserver(targetNamespaceName string) configobserver.ObserveConfigFunc {
+	cloudObserver := &cloudProviderObserver{targetNamespaceName: targetNamespaceName}
+	return cloudObserver.ObserveCloudProviderNames
+}
+
+type cloudProviderObserver struct {
```
Comment (deads2k) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: needs the path values

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	targetNamespaceName     string
+	cloudProviderNamePath   []string
+	cloudProviderConfigPath []string
+}
+
+// ObserveCloudProviderNames observes the cloud provider from the global cluster infrastructure resource.
+func (cloud *cloudProviderObserver) ObserveCloudProviderNames(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```
Comment (mfojtik) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: nit: haven't seen `cloud` too much, we usually use `c` ;-)

**Comment by gnufied** [comment]
proof that it works - https://github.com/openshift/cluster-kube-controller-manager-operator/pull/217

**Pull request merged**